File: RKGeonames/DemographicData.m
//
//  DemographicData.m
//  RKGeonames
//
//  Created by Stefan Burettea on 13/09/2013.
//  Copyright (c) 2013 Stefan Burettea. All rights reserved.
//

#import "DemographicData.h"

@implementation DemographicData

// |+|=======================================================================|+|
// |+|                                                                       |+|
// |+|    FUNCTION NAME: initWithTotalPopulation                             |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    DESCRIPTION:                                                       |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    PARAMETERS:                                                        |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    RETURN VALUE:  N/A                                                 |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|=======================================================================|+|
- (id)initWithTotalPopulation:(NSString *)totalPopulation
             populationGrowth:(NSString *)populationGrowth
                    birthRate:(NSString *)birthRate
                    deathRate:(NSString *)deathRate;
{
    self = [super init];
    if(self)
    {
        _totalPopulation = totalPopulation;
        _populationGrowth = populationGrowth;
        _birthRate = birthRate;
        _deathRate = deathRate;
    }
    
    return self;
}

// |+|=======================================================================|+|
// |+|                                                                       |+|
// |+|    FUNCTION NAME: emptyDemographicData                                |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    DESCRIPTION:                                                       |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    PARAMETERS:                                                        |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|    RETURN VALUE:  N/A                                                 |+|
// |+|                                                                       |+|
// |+|                                                                       |+|
// |+|=======================================================================|+|
+ (id)data;
{
    return [[self alloc] initWithTotalPopulation:@"N/A"
                                           populationGrowth:@"N/A"
                                                  birthRate:@"N/A"
                                                  deathRate:@"N/A"];
    
}


@end

File: CountriesOfTheWorld/DemographicData.swift
//
//  DemographicData.swift
//  RKGeonames
//
//  Created by Stefan Buretea on 9/21/14.
//  Copyright (c) 2014 Stefan Burettea. All rights reserved.
//

import Foundation

class DemographicData : NSObject {
    
    let totalPopulation: NSString
    let populationGrowth: NSString
    let birthRate: NSString
    let deathRate: NSString
    
    init(totalPopulation: NSString, populationGrowth: NSString, birthRate: NSString, deathRate: NSString) {
        
        self.totalPopulation = totalPopulation
        self.populationGrowth = populationGrowth
        self.birthRate = birthRate
        self.deathRate = deathRate
    }
    
    class func data() -> DemographicData {
        return DemographicData(totalPopulation: "" as NSString!, populationGrowth: "" as NSString!, birthRate: "" as NSString!, deathRate: "" as NSString!)
    }
}